core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -695,6 +695,126 @@ class DAGSchedulerSuite
     assertDataStructuresEmpty()
   }
 
+
+  // Helper function to validate state when creating tests for task failures
+  def checkStageId(stageId: Int, attempt: Int, stageAttempt: TaskSet) {
+    assert(stageAttempt.stageId === stageId,
+      s": expected stage $stageId, got ${stageAttempt.stageId}")
+    assert(stageAttempt.stageAttemptId == attempt,
+      s": expected stage attempt $attempt, got ${stageAttempt.stageAttemptId}")
+  }
+
+  def makeCompletions(stageAttempt: TaskSet, reduceParts: Int): Seq[(Success.type, MapStatus)] = {
+    stageAttempt.tasks.zipWithIndex.map { case (task, idx) =>
+      (Success, makeMapStatus("host" + ('A' + idx).toChar, reduceParts))
+    }.toSeq
+  }
+
```

Comment:
thanks Mark -- these are the utils I'm stealing from https://github.com/apache/spark/pull/5636 (where these reduce a *lot* of noise), might be best to move this discussion there?  Then again this pr is probably the easiest to merge, so maybe we can do this here first?

I don't particularly care either way, just want to avoid a fractured discussion (which would be my fault for doing this ...)